Source/Extensions/Blazorise.LoadingIndicator/LoadingIndicator.razor.cs

```diff
@@ -0,0 +1,312 @@
+    /// </summary>
+    public partial class LoadingIndicator : BaseComponent, IDisposable
+    {
+        #region Members
+
+        private ClassBuilder screenClasses;
+        private ClassBuilder indicatorClasses;
+        private StyleBuilder screenStyles;
+        private StyleBuilder indicatorStyles;
+
+        private LoadingIndicatorService service;
+
+        private bool? loaded;
+        private bool loadedParameter = true;
+
+        private bool? busy;
+        private bool busyParameter;
+
+        #endregion
+
+        #region Methods
+
+        public LoadingIndicator()
+        {
+            screenClasses = new( BuildScreenClasses );
+            screenStyles = new( BuildScreenStyles );
+            indicatorClasses = new( BuildIndicatorClasses );
+            indicatorStyles = new( BuildIndicatorStyles );
+        }
+
+        protected override void BuildClasses( ClassBuilder builder )
+        {
+            builder.Append( "bl-wrapper" );
+            base.BuildClasses( builder );
+        }
+
+        protected override void BuildStyles( StyleBuilder builder )
+        {
+            builder.Append( "position:relative", Busy && !FullScreen );
+            builder.Append( "display:inline-block", Inline );
+            base.BuildStyles( builder );
+        }
+        private void BuildScreenClasses( ClassBuilder builder )
+        {
+            builder.Append( "bl-overlay" );
+            builder.Append( FullScreen ? "bl-overlay-fixed" : "bl-overlay-relative" );
+            base.BuildClasses( builder );
+        }
+        private void BuildIndicatorClasses( ClassBuilder builder )
+        {
+            builder.Append( "bl-overlay" );
+            builder.Append( FullScreen ? "bl-overlay-fixed" : "bl-overlay-relative" );
+            builder.Append( "bl-indicator" );
+            base.BuildClasses( builder );
+        }
+
+        private void BuildScreenStyles( StyleBuilder builder )
+        {
+            builder.Append( $"opacity:{ScreenOpacity}" );
+            builder.Append( $"background-color:{ScreenColor}" );
+            builder.Append( $"z-index:{ZIndex}", ZIndex.HasValue );
+            base.BuildStyles( builder );
+        }
+
+        private void BuildIndicatorStyles( StyleBuilder builder )
+        {
+            builder.Append( $"z-index:{ZIndex}", ZIndex.HasValue );
+            builder.Append( $"justify-content:start", IndicatorHorizontalPlacement == Placement.Start );
+            builder.Append( $"justify-content:end", IndicatorHorizontalPlacement == Placement.End );
+            builder.Append( $"align-items:start", IndicatorVerticalPlacement == Placement.Top );
+            builder.Append( $"align-items:end", IndicatorVerticalPlacement == Placement.Bottom );
+            builder.Append( $"padding:{IndicatorMargin}", !string.IsNullOrEmpty( IndicatorMargin ) );
+            base.BuildStyles( builder );
+        }
+
+        private void DirtyClassesAndStyles()
```

Comment:
Override from base class rather than adding new method

```
protected internal override void DirtyClasses()
{
    IndicatorClassBuilder.Dirty();

    base.DirtyClasses();
}
```